components/qma6100p/include/qma6100p.h

```diff
@@ -0,0 +1,352 @@
+ *
+ * @param port I2C port number
+ * @param dev_addr I2C device address of sensor
+ *
+ * @return
+ *     - NULL Fail
+ *     - Others Success
+ */
+qma6100p_handle_t qma6100p_create(i2c_port_t port, const uint16_t dev_addr);
+
+/**
+ * @brief Delete and release a sensor object
+ *
+ * @param sensor object handle of qma6100p
+ */
+void qma6100p_delete(qma6100p_handle_t sensor);
+
+/**
+ * @brief Get device identification of qma6100p
+ *
+ * @param sensor object handle of qma6100p
+ * @param deviceid a pointer of device ID
+ *
+ * @return
+ *     - ESP_OK Success
+ *     - ESP_FAIL Fail
+ */
+esp_err_t qma6100p_get_deviceid(qma6100p_handle_t sensor, uint8_t *const deviceid);
+
+/**
+ * @brief Load non volatile memory
+ *
+ * @param sensor object handle of qma6100p
+ *
+ * @return
+ *     - ESP_OK Success
+ *     - ESP_FAIL Fail
+ */
+esp_err_t qma6100p_nvm_load(qma6100p_handle_t sensor);
+
+/**
+ * @brief Wake up qma6100p
+ *
+ * @param sensor object handle of qma6100p
+ *
+ * @return
+ *     - ESP_OK Success
+ *     - ESP_FAIL Fail
+ */
+esp_err_t qma6100p_wake_up(qma6100p_handle_t sensor);
+
+/**
+ * @brief Enter sleep mode
+ *
+ * @param sensor object handle of qma6100p
+ *
+ * @return
+ *     - ESP_OK Success
+ *     - ESP_FAIL Fail
+ */
+esp_err_t qma6100p_sleep(qma6100p_handle_t sensor);
+
+/**
+ * @brief Set accelerometer full scale range
+ *
+ * @param sensor object handle of qma6100p
+ * @param acce_fs accelerometer full scale range
+ *
+ * @return
+ *     - ESP_OK Success
+ *     - ESP_FAIL Fail
+ */
+esp_err_t qma6100p_config(qma6100p_handle_t sensor, const qma6100p_acce_fs_t acce_fs);
+
+/**
+ * @brief Get accelerometer sensitivity
+ *
+ * @param sensor object handle of qma6100p
+ * @param acce_sensitivity accelerometer sensitivity
+ *
+ * @return
+ *     - ESP_OK Success
+ *     - ESP_FAIL Fail
+ */
+esp_err_t qma6100p_get_acce_sensitivity(qma6100p_handle_t sensor, float *const acce_sensitivity);
+
+/**
+ * @brief Configure INT pin behavior and setup target GPIO.
+ *
+ * @warning Not tested, implemented according to datasheet.
+ *
+ * @param sensor object handle of qma6100p
+ * @param interrupt_configuration qma6100p INT pin configuration parameters
+ *
+ * @return
+ *      - ESP_OK Success
+ *      - ESP_ERR_INVALID_ARG A parameter is NULL or incorrect
+ *      - ESP_FAIL Failed to configure INT pin on qma6100p
+ */
+esp_err_t qma6100p_config_interrupt(qma6100p_handle_t sensor, int int_num, const qma6100p_int_config_t *const interrupt_configuration);
+
+/**
+ * @brief Register an Interrupt Service Routine to handle qma6100p interrupts.
+ *
+ * @param sensor object handle of qma6100p
+ * @param isr function to handle interrupts produced by qma6100p
+ *
+ * @return
+ *      - ESP_OK Success
+ *      - ESP_ERR_INVALID_ARG A parameter is NULL or not valid
+ *      - ESP_FAIL Failed to register ISR
+ */
+esp_err_t qma6100p_register_isr(qma6100p_handle_t sensor, const qma6100p_isr_t isr);
```

Comment:
Do we need this function when we pass the isr inside of ` qma6100p_int_config_t->isr` ?

Reply:
Acts only as a helper function, removed from qma6100p.h file.